### PR TITLE
Fix FCM token and UP endpoint inconsistency

### PR DESCRIPTION
**Pull request by p1gp1g**
## Type of change

- [ ] Feature
- [x] Bugfix
- [ ] Technical
- [ ] Other :

## Content

* Bump embedded_fcm_distributor to 2.1.3 which fix a potential issue with fcmToken
* Force-delete the FCM token during the troubleshooting list : 
    * It acts like fcm has never been initialized (like data wipe) and make sure the new token is registered

This PR force the app to have a new FCM token during the troubleshoot tests, like if the app were re-installed. This solution is a bit rough, and recent changes have probably fix fcm issues (I'd still bump the embedded_distributor). But this PR make it easier for users to fix their notifications if the issues are still present.

## Motivation and context

There are many issue with push notifications like : 
https://github.com/vector-im/element-android/issues/6572
https://github.com/vector-im/element-android/issues/6975
https://github.com/vector-im/element-android/issues/6379

Some users even wipe the app data (or re-install element).

## Tested devices

- [x] Physical
- [ ] Emulator
- OS version(s): 12

## Checklist

- [ ] Changes has been tested on an Android device or Android emulator with API 21
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#accessibility
- [x] Pull request is based on the develop branch
- [x] Pull request includes a new file under ./changelog.d. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#changelog
- [ ] Pull request includes screenshots or videos if containing UI changes
- [x] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [x] You've made a self review of your PR
- [ ] If you have modified the screen flow, or added new screens to the application, you have updated the test [UiAllScreensSanityTest.allScreensTest()](https://github.com/vector-im/element-android/blob/main/vector/src/androidTest/java/im/vector/app/ui/UiAllScreensSanityTest.kt#L73)

### CODE REVIEW

File: vector/src/gplay/java/im/vector/app/gplay/features/settings/troubleshoot/TestFirebaseToken.kt [hunk 1]
```diff
@@ -36,43 +37,66 @@ class TestFirebaseToken @Inject constructor(
...
-                                    stringProvider.getString(R.string.settings_troubleshoot_test_fcm_failed_service_not_available, errorMsg)
-                                }
-                                "TOO_MANY_REGISTRATIONS" -> {
-                                    stringProvider.getString(R.string.settings_troubleshoot_test_fcm_failed_too_many_registration, errorMsg)
-                                }
-                                "ACCOUNT_MISSING" -> {
-                                    quickFix = object : TroubleshootQuickFix(R.string.settings_troubleshoot_test_fcm_failed_account_missing_quick_fix) {
-                                        override fun doFix() {
-                                            startAddGoogleAccountIntent(context, activityResultLauncher)
+            FirebaseMessaging.getInstance().deleteToken()
```
Comment (bmarty) [inline vector/src/gplay/java/im/vector/app/gplay/features/settings/troubleshoot/TestFirebaseToken.kt:61]: The test itself should not modify the environment. Only the quick fix can do that. Is it possible to move this to the quick fix lambda, or even maybe create a dedicated Test?

File: vector/src/gplay/java/im/vector/app/gplay/features/settings/troubleshoot/TestFirebaseToken.kt [hunk 2]
```diff
@@ -36,43 +37,66 @@ class TestFirebaseToken @Inject constructor(
...
+    //
+    // That's a bit hacky
+    private fun forceRemoveFCMToken(context: Context) {
+        listOf(
+                "com.google.firebase.messaging",
+                "com.google.android.gms.appid"
+        ).map {
+            context.getSharedPreferences(it, Context.MODE_PRIVATE)
+        }.forEach { prefs ->
+            prefs.all.forEach {
```
Comment (bmarty) [inline vector/src/gplay/java/im/vector/app/gplay/features/settings/troubleshoot/TestFirebaseToken.kt:52]: could simply use `prefs.edit { it.clear() }`.

**Review by bmarty — COMMENTED** [review]
Thanks for the PR.
This looks a bit hacky, as you said. Previously (I mean without using UnifiedPush), it was not necessary to do that. Please see my other remarks.

**Comment by p1gp1g** [comment]
Well, I should have started to explain a bit (and more points since your review):
1. Up to version 2.1.3, there were an issue where the app could be in inconsistent state : when `onNewToken` (from FirebaseService) was not called/or finished : it was possible that the fcmToken was not sent as a new "endpoint" and then the pusher was not updated
2. While debugging, I've found some states, unrelated to unifiedpush, where the fcmToken (received with `FirebaseMessaging.getInstance().token`) was not working : probably a ratelimite or a timestamp (or a problem with microg)
3. That's why, given **2.**, I wanted to modify the environment during this tests instead of the fix
4. From your review and given **1.**, I wanted to do this *hacky* pref edit only when the pusher key was different from the fcmToken and I've seen there were another inconsistency that, if it weren't there, would have fix the issue from the embedded distributor (cf 4a79230346)

**Comment by p1gp1g** [comment]
I will come with a better solution

**Pull request closed without merging**